src/main.cpp

```diff
@@ -167,12 +167,15 @@ static std::string parse_commandline(int argc, char *argv[]) {
 
     if (vm.count("seed")) {
         cfg_rng_seed = vm["seed"].as<std::uint64_t>();
+        if (cfg_rng_seed == 0) {
+          throw std::runtime_error("RNG seed cannot be 0");
+        }
         if (cfg_num_threads > 1) {
```

Comment:
I think this is good for the case where threads is using the default. But how about adding a check for vm.count("threads") and error when the user is explicitly trying to set more than one thread.